sources/packages/web/src/components/students/StudentAssessmentDetails.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <request-assessment
+    :applicationId="applicationId"
+    :showWhenEmpty="false"
+    @viewStudentAppeal="goToStudentAppeal"
+  />
+  <history-assessment
+    :applicationId="applicationId"
+    :viewRequestTypes="studentAssessmentRequestTypes"
+    @viewStudentAppeal="goToStudentAppeal"
+    @viewAssessment="gotToViewAssessment"
+  />
+</template>
+<script lang="ts">
+import { defineComponent, ref } from "vue";
+import { useRouter } from "vue-router";
+import { AssessmentTriggerType } from "@/types";
+import { StudentRoutesConst } from "@/constants/routes/RouteConstants";
+import HistoryAssessment from "@/components/aest/students/assessment/History.vue";
+import RequestAssessment from "@/components/aest/students/assessment/Request.vue";
+export default defineComponent({
+  components: {
+    HistoryAssessment,
+    RequestAssessment,
+  },
+  props: {
+    applicationId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props) {
+    const router = useRouter();
+    // The assessment trigger types for which the request form is visible to student.
+    const studentAssessmentRequestTypes = ref<AssessmentTriggerType[]>([
```

Comment:
looks like it is only used in the history table, can't we add this logic in the history component?

Reply:
This object must be residing at the consumer end of the component to choose what are the trigger types which are available to show request form. 

Every time parent doesn't have to provide it as the history component will show all possible request form on default.
Only if parent chooses to show only selected trigger types then they will have to supply the values.

Reply:
I believe that it does not need to be declared reactive since its value will never change. It can also be added directly to the template (but either way is fine).